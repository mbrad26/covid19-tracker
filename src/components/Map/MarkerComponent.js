import React from 'react';
import { Marker} from 'react-map-gl';

import { stats, style } from './utils';
import PopupComponent from '../../containers/PopupComponent.js';

const MarkerComponent = ({ countries, activeCountry, onSetActive }) => {

  console.log('COUNTRY MARKER: ', activeCountry);

  return (
    <>
      {countries.map(country => {
        const cases = stats(country.cases)

        return (
            <Marker
              key={country.country}
              latitude={country.countryInfo.lat}
              longitude={country.countryInfo.long}
            >
              <button
                id='marker'
                style={style[cases]}
                onClick={e => {
                  e.preventDefault();
                  // setActive(country);
                  onSetActive(country);
                }}
              >
              </button>
            </Marker>
          )
        })
      }
      <PopupComponent />
    </>
  );
};

export default MarkerComponent;
