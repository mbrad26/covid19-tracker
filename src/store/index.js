import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers';
import rootSaga from '../sagas';

const logger = createLogger();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  applyMiddleware(saga, logger)
);

// console.log('STORE: ', store.getState());

saga.run(rootSaga);

export default store;
