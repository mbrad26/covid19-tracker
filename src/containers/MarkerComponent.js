/* istanbul ignore file */
import { connect } from 'react-redux';

import { getData} from '../selectors/data';
import { doSetActive } from '../actions/countryData';
import MarkerComponent from '../components/Map/MarkerComponent';

const mapStateToProps = ({ countriesDataState, countryDataState }) => ({
  countries: getData(countriesDataState),
});

const mapDispatchToProps = dispatch => ({
  onSetActive: data => dispatch(doSetActive(data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(MarkerComponent);
