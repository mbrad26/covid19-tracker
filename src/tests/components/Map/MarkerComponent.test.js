import React from 'react';
import { Provider } from 'react-redux';
import { render, screen, waitForElement } from '@testing-library/react';

import store from '../../../store/';
import { unsortedData, country } from '../../fixtures';
import { Marker } from 'react-map-gl';
import MarkerComponent from '../../../containers/MarkerComponent';

describe('MarkerComponent', () => {
  const markerComponentProps = {
    countries: unsortedData,
    onSetActive: jest.fn(),
  };

  const markerProps = {
    key: country.country,
    latitude: country.countryInfo.lat,
    longitude: country.countryInfo.long,
  };

  it('renders a snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <MarkerComponent {...markerComponentProps} >
          <Marker {...markerProps} />
        </MarkerComponent>
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

});
