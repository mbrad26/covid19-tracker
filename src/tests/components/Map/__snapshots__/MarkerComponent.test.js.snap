// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkerComponent renders a snapshot 1`] = `
<div
  id="popup"
/>
`;
